### DISCUSSION TITLE
Problems with using STL models

### DISCUSSION THREAD

**Discussion by TONGLoki** [surgical-robotics-ai/surgical_robotics_challenge#77] [Q&A] [answered]
How to correctly import the model in STL format. After I imported the STL file, the simulation became very stuttery, and besides, the model I imported was very dark, even if I turned up the brightness to 1, it was still very dark. 

In addition, can I replace the head of the gripper of the PSM separately, the gripper in the simulation seems to be different from the reality.

Is it possible to collect data directly from the PSM action in the example (i.e., the actual device operation during demonstration)?

**Comment by adnanmunawar** [comment, marked as answer]
Hello, to import a new rigid body in the scene, you could use the [blender ambf addon](https://github.com/WPI-AIM/ambf_addon). The mesh itself may be too high resolution for collision, so you would need to approximate via primitive shapes, or reduced verticies mesh. These can be accomplished via the Blender ambf addon.

For changing the gripper, again you can use the addon.

What kind of data do you want to collect? You can use ROS bag to collect any data returned from the simulator.
